core/blockchain.go

```diff
@@ -1552,7 +1552,11 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 						bc.triegc.Push(root, number)
 						break
 					}
-					go triedb.Dereference(root.(common.Hash))
+					wg.Add(1)
+					go func() {
+						triedb.Dereference(root.(common.Hash))
+						wg.Done()
```

Comment:
Do we need to add this lock? Since GC can be done in background.
And in fact there is already another `wg.Wait()` in https://github.com/bnb-chain/bsc/blob/master/core/state/statedb.go#L1495 to make sure GC will be done in `StateDB.Commit()`

I think it is irrelevant to the invalid merkle tree, right?

Reply:
sorry, my fault, there is no wait in `StateDB.Commit` for `Dereference`.
you are right, this PR do solve the UT fail.